src/rp2_common/hardware_clocks/include/hardware/clocks.h

```diff
@@ -354,9 +354,26 @@ void clocks_enable_resus(resus_callback_t resus_callback);
  * \param gpio The GPIO pin to output the clock to. Valid GPIOs are: 21, 23, 24, 25. These GPIOs are connected to the GPOUT0-3 clock generators.
  * \param src  The source clock. See the register field CLOCKS_CLK_GPOUT0_CTRL_AUXSRC for a full list. The list is the same for each GPOUT clock generator.
  * \param div_int  The integer part of the value to divide the source clock by. This is useful to not overwhelm the GPIO pin with a fast clock. this is in range of 1..2^24-1.
- * \param div_frac The fractional part of the value to divide the source clock by. This is in range of 0..255 (/256).
+ * \param div_frac16 The fractional part of the value to divide the source clock by. This is in range of 0..65536 (/65536).
  */
-void clock_gpio_init_int_frac(uint gpio, uint src, uint32_t div_int, uint8_t div_frac);
+void clock_gpio_init_int_frac16(uint gpio, uint src, uint32_t div_int, uint16_t div_frac16);
```

Comment:
If we're changing the name of the function anyway to specify the width of the fractional part, _might_ it also be worth specifying the width of the integer part too i.e. `clock_gpio_init_int24_frac16` ? Or perhaps that is just unnecessarily overly-specific.

Reply:
God no :)

Reply:
yeah, the reason for the number with the frac is that it affects the meaning of value N (e.g. is it 37/255 or 37/65535)... for the integer parts N means N

Reply:
> for the integer parts N means N

Oh of course; I must have been having a brain-fart! :laughing: